pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -643,6 +643,38 @@ func adsClientCount() int {
 	return n
 }
 
+func (s *DiscoveryServer) ProxyUpdate(clusterID, ip string) {
+	var connection *XdsConnection
+
+	adsClientsMutex.RLock()
+	// Create a temp map to avoid locking the add/remove
+	for _, v := range adsClients {
+		if v.modelNode.ClusterID == clusterID && v.modelNode.IPAddresses[0] == ip {
```

Comment:
nit for future: cant we have a map of ip-cluster: connection ?

Reply:
Yes, first we need the clusterID for the node.